artifactory/commands/transferfiles/manager.go

```diff
@@ -272,6 +272,9 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 
 		// Each uploading thread receive a token and a node id from the source via the uploadChunkChan, so this go routine can poll on its status.
 		fillChunkDataBatch(&chunksLifeCycleManager, uploadChunkChan)
+		if err := chunksLifeCycleManager.StoreStaleChunks(phaseBase.stateManager); err != nil {
+			log.Error("Couldn't store the stale chunks:", err.Error())
+		}
```

Comment:
Since the stale chunks are at least 30 minutes old, I think it is really unnecessary to call the `StoreStaleChunks` function every 3 seconds. It may be sufficient to call this function once every 60 cycles of this loop (which is approximately once every 3 minutes). This will save redundant processing.